jcommune-service/src/main/java/org/jtalks/jcommune/service/transactional/TransactionalPostService.java

```diff
@@ -149,11 +150,14 @@ public void deletePost(Post post) {
             branch.clearLastPost();
         }
 
-
         // todo: event API?
         topicDao.saveOrUpdate(topic);
         securityService.deleteFromAcl(post);
-        notificationService.subscribedEntityChanged(topic);
+
+        Set<JCUser> topicSubscribers = new HashSet<>(topic.getSubscribers());
+        topicSubscribers.remove(user);
+        notificationService.subscribedEntityChanged(topic, topicSubscribers);
```

Comment:
Agree with @vps2 - the user may be unsubscribed. We may introduce a method `subscriptionService.entityChanged(entity, userToNotify)`, I'm just thinking if this will make things more unclear since with that we'll be finding subscribers in different places on different layers.